### PR TITLE
EPO-2757 Create basic brightness vs Distance plot

### PR DESCRIPTION
**Pull request by georgegomez144**

### CODE REVIEW

File: src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      :global {
+        .md-tile-text--primary, .md-tile-text--primary.md-text--theme-primary, .md-text {
+          /* stylelint-disable-next-line declaration-no-important */
+          color: $white !important;
+        }
+      }
+    }
+  }
+}
+
```
Comment (blnkt) [inline src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss]: We will get to the point of needing this plot to play nice with our Drawer layout, but right now none of these styles are applicable so either delete or comment out for now.

File: src/containers/BrightnessVsDistanceContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    });
+  }
+
+  Callback = d => {
+    console.log(d); // eslint-disable-line no-console
+    // const { activeQuestionId } = this.props;
+    // if (activeQuestionId) {
+    //   updateAnswer(activeQuestionId, activeSupernova);
+    // }
+  };
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/containers/BrightnessVsDistanceContainer.jsx]: I was planning on using userBrightnessVsDistanceCallback as that handler. Removing the Callback()

File: src/components/charts/brightnessVsDistance/PointLabel.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { getElDims } from '../../../lib/utilities.js';
+import styles from './brightness-vs-distance.module.scss';
+
+class PointLabel extends React.PureComponent {
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/PointLabel.jsx]: removed.

File: src/components/charts/brightnessVsDistance/Point.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          stroke="transparent"
+          tabIndex="0"
+        />
+        {label && x && y && (
+          <PointLabel
+            {...{ label, selected, hovered }}
+            x={x + this.baseDiameter + this.baseRadius}
+            y={y + this.baseRadius}
+          />
+        )}
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/Point.jsx]: removed.

File: src/components/charts/brightnessVsDistance/Point.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { select as d3Select } from 'd3-selection';
+import { easeElastic as d3EaseElastic } from 'd3-ease';
+import PointLabel from './PointLabel';
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/Point.jsx]: removed.

File: src/components/charts/brightnessVsDistance/Points.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+          const x = d[xValueAccessor];
+          const y = d[yValueAccessor];
+          const selected = includes(selectedData, d);
+          const hovered = includes(hoveredData, d);
+          const classes = classnames(`data-point-${name}`, 'data-point', {
+            [pointClasses]: pointClasses,
+            selected,
+            hovered,
+            [notActive]: (selectedData || hoveredData) && !selected && !hovered,
+            [invisible]: x === null || y === null,
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/Points.jsx]: removed.

File: src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  text-anchor: middle;
+}
+
+.galaxy-point {
+  stroke: none;
+  transition: stroke $timing $duration, opacity $timing $duration;
+
+  &.invisible {
+    opacity: 0;
+  }
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss]: removed.

File: src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+.point-label {
+  @include labelPrimary;
+
+  &.active-label {
+    font-weight: $medium;
+  }
+}
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss]: removed.

File: src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss (hunk 1 above)
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/brightness-vs-distance.module.scss]: commented out.

File: src/components/charts/brightnessVsDistance/brightnessVsDistanceUtilities.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import isEmpty from 'lodash/isEmpty';
+
+export const emptyUserData = data => {
+  if (!data) return data;
+
+  return data.map(galaxy => {
+    return { ...galaxy, distance: null, velocity: null };
+  });
+};
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/brightnessVsDistance/brightnessVsDistanceUtilities.js]: deleted.

File: src/components/charts/shared/Tooltip.jsx [hunk 10]
```diff
@@ -92,18 +96,19 @@ class Tooltip extends React.PureComponent {
...
   }
 
-  renderAccessor(accessor, label, data) {
+  renderAccessor(accessor, label, data, unit) {
     return (
       <div className="value-row" key={accessor}>
         <span>{label || capitalize(accessor)}: </span>
-        {data.length > 1 && this.renderRange(accessor, data)}
-        {data.length === 1 && this.renderValue(accessor, data[0][accessor])}
+        {data.length > 1 && this.renderRange(accessor, data, unit)}
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/shared/Tooltip.jsx]: you're right. completely missed this.

File: src/components/charts/shared/Tooltip.jsx [hunk 11]
```diff
@@ -92,18 +96,19 @@ class Tooltip extends React.PureComponent {
...
+        {data.length === 1 &&
+          this.renderValue(accessor, data[0][accessor], unit)}
       </div>
     );
   }
 
   render() {
-    const { data, accessors, labels } = this.props;
+    const { data, accessors, units, labels } = this.props;
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/shared/Tooltip.jsx]: i decided to use the `defaultProps...unit: []`. The need to use this empty array prop is needed for this component.

File: src/components/charts/shared/unit/StandardUnit.jsx [hunk 12]
```diff
@@ -4,15 +4,15 @@ import { renderUnit } from './utilities.js';
...
 };
 
-StellarUnit.propTypes = {
+StandardUnit.propTypes = {
   type: PropTypes.string,
   isSvg: PropTypes.bool,
 };
 
-export default StellarUnit;
+export default StandardUnit;
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/components/charts/shared/unit/StandardUnit.jsx]: 👍

File: src/containers/BrightnessVsDistanceContainer.jsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+BrightnessVsDistanceContainer.propTypes = {
+  data: PropTypes.object,
+  options: PropTypes.object,
+  widget: PropTypes.object,
+  answers: PropTypes.object,
+  activeQuestionId: PropTypes.string,
+  activeAnswer: PropTypes.object,
+  updateAnswer: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (georgegomez144) [reply src/containers/BrightnessVsDistanceContainer.jsx]: removed.

**Pull request merged**